test/powershell/Modules/Microsoft.PowerShell.Utility/JsonObject.Tests.ps1

```diff
@@ -1,46 +1,47 @@
-            $InputObject,
-            [Parameter(Position = 0)]
-            $ExpectedException
-        )
-
-        try
-        {
-            & $InputObject
-            throw "Should throw exception"
-        }
-        catch
-        {
-            $_.FullyQualifiedErrorId | should be $ExpectedException
-        }
-    }
-
     $validStrings = @(
-        @{ name = "empty"; str = "" }
-        @{ name = "spaces"; str = "  " }
-        @{ name = "object"; str = "{a:1}" }
+        @{ name = "null";   str = $null;   ReturnHashTable = $true  }
+        @{ name = "empty";  str = "";      ReturnHashTable = $true  }
+        @{ name = "spaces"; str = "  ";    ReturnHashTable = $true  }
+        @{ name = "object"; str = "{a:1}"; ReturnHashTable = $true  }
+        @{ name = "null";   str = $null;   ReturnHashTable = $false }
+        @{ name = "empty";  str = "";      ReturnHashTable = $false }
+        @{ name = "spaces"; str = "  ";    ReturnHashTable = $false }
+        @{ name = "object"; str = "{a:1}"; ReturnHashTable = $false }
     )
 
-    It 'no error for valid string - <name>' -TestCase $validStrings {
-        param ($str)
+    It 'no error for valid string ''<name>'' with -ReturnHashTable:$<ReturnHashTable>' -TestCase $validStrings {
+        param ($str, $ReturnHashTable)
         $errRecord = $null
-        [Microsoft.PowerShell.Commands.JsonObject]::ConvertFromJson($str, [ref]$errRecord)
+        [Microsoft.PowerShell.Commands.JsonObject]::ConvertFromJson($str, $ReturnHashTable, [ref]$errRecord)
         $errRecord | Should BeNullOrEmpty
     }
 
     $invalidStrings = @(
-        @{ name = "plain text"; str = "plaintext" }
-        @{ name = "part"; str = '{"a" :' }
+        @{ name = "plain text"; str = "plaintext"; ReturnHashTable = $true  }
+        @{ name = "part";       str = '{"a" :';    ReturnHashTable = $true  }
+        @{ name = "plain text"; str = "plaintext"; ReturnHashTable = $false }
+        @{ name = "part";       str = '{"a" :';    ReturnHashTable = $false }
     )
 
-    It 'throw ArgumentException for invalid string - <name>' -TestCase $invalidStrings  {
-        param ($str)
+    It 'throw ArgumentException for invalid string ''<name>'' with -ReturnHashTable:$<ReturnHashTable>' -TestCase $invalidStrings  {
+        param ($str, $ReturnHashTable)
         $errRecord = $null
-        { [Microsoft.PowerShell.Commands.JsonObject]::ConvertFromJson($str, [ref]$errRecord) } | ShouldThrow 'ArgumentException'
+        { [Microsoft.PowerShell.Commands.JsonObject]::ConvertFromJson($str, $ReturnHashTable, [ref]$errRecord) } | ShouldBeErrorId "ArgumentException"
+    }
+
+    $jsonWithEmptyKey = '{"": "Value"}'
+    It 'throw InvalidOperationException when json contains empty key name' {
+        $errorRecord = $null
+        [Microsoft.PowerShell.Commands.JsonObject]::ConvertFromJson($jsonWithEmptyKey, [ref]$errorRecord)
+        $errorRecord.Exception.GetType() | Should Be System.InvalidOperationException
```

Comment:
Usually we don't check an exception types. Please remove.
(For type check we use Should BeOfType)